add dsb to SeuratWrappers

MattPM opened the pull request:
dsb is an R package for normalizing ADT data. 

manuscript: https://www.nature.com/articles/s41467-022-29356-8
CRAN:  https://CRAN.R-project.org/package=dsb
github: https://github.com/niaid/dsb

We would like to add dsb to Seurat wrappers. Please let us know if we correctly added the link to the vignette build from CI in readme.md (see input to readme added below with the htmlpreview.github.io based html link. Let us know if anything else is required. 

Thank you! 

| dsb | [Normalize and denoise ADT protein data using dsb with Seurat](http://htmlpreview.github.io/?https://github.com/satijalab/seurat-wrappers/blob/master/docs/dsb.html) | Mulè et al., Nature Communications 2022 | https://CRAN.R-project.org/package=dsb |

mojaveazure commented on R/dsb.R:
```diff
@@ -0,0 +1,515 @@
+#'     }
+#'   }
+#'   # apply quantile clipping of outliers
+#'   if (isTRUE(quantile.clipping)) {
+#'     ql = apply(norm_adt, 1, FUN = stats::quantile, quantile.clip[1])
+#'     qh = apply(norm_adt, 1, FUN = stats::quantile, quantile.clip[2])
+#'     for (i in 1:nrow(norm_adt)) {
+#'       norm_adt[i, ] = ifelse(norm_adt[i, ] < ql[i], ql[i], norm_adt[i, ])
+#'       norm_adt[i, ] = ifelse(norm_adt[i, ] > qh[i], qh[i], norm_adt[i, ])
+#'     }
+#'   }
+#'   # write internal stats to R list to return if return.stats = TRUE
+#'   if(isTRUE(return.stats)) {
+#'     print('returning stats and results in a list; access normalized matrix with x$dsb_normalized_matrix')
+#'     protein_stats = list(
+#'
+#'       'raw cell matrix stats' = data.frame(
+#'         cell_mean = apply(adt_log, 1 , mean),
+#'         cell_sd = apply(adt_log, 1 , sd)),
+#'
+#'       'dsb normalized matrix stats' =
+#'         data.frame(
+#'           dsb_mean = apply(norm_adt, 1 , mean),
+#'           dsb_sd = apply(adt_log, 1 , sd))
+#'     )
+#'     if(isTRUE(denoise.counts)) {
+#'       technical_stats = cbind(t(noise_matrix), dsb_technical_component = noise_vector)
+#'     } else{
+#'       technical_stats = NULL
+#'     }
+#'     ret_obj = list(
+#'       'dsb_normalized_matrix' = norm_adt,
+#'       'technical_stats' = technical_stats,
+#'       'protein_stats' = protein_stats
+#'     )
+#'     return(ret_obj)
+#'   } else {
+#'     return(norm_adt)
+#'   }
+#' }
```
I'm not sure what's going on here; it seems like all the code has been commented out except for an odd `source` call. All code in `R/internal.R` are available to other functions within SeuratWrappers so you shouldn't need the `source`

MattPM replied:
Thanks for the note-for efficiency, in our workflow the 2 core functions are implemented just upstream of creation of a Seurat object using the dsb package from CRAN directly (this prevents Seurat object from having to access a potentially very large empty droplet matrix–see vignette). My thought was to have users load dsb package from CRAN directly then follow the seurat wrapper vignette which uses standard Seurat functions. This would also allow me to only maintain the CRAN pacage instead of having to push future patches to CRAN and SeuratWrappers. However, the functions are quite mature at this point. I'm happy to just include them in SeuratWrappers (the source call from before was just to make sure the user installed the package from CRAN). I will push code for functions directly.

MattPM commented:
Thanks for your note @mojaveazure please see comment and latest commit. 
Best, 
-Matt

MattPM commented:
Hi @mojaveazure 

I addressed your comment in the commit above https://github.com/satijalab/seurat-wrappers/pull/130/commits/75647262f01acf9c4365d3cd4bf68a65ecb72239. Functions from the package are now included. 

Wanted to check in and see if you're waiting for anything else on my end?  

Please let me know and thanks again.

MattPM commented:
Hello @mojaveazure
Please see comments above. I addressed your previous comment in the commit above. Functions from the package are now included.

Are you waiting for anything else on my end? Many thanks.
